test/FastTests/Client/QueriesWithCustomFunctions.cs

```diff
@@ -1667,7 +1668,230 @@ select new
+                        Deleted = false,
+                        SubDocuments = new List<Document>
+                    {
+                        new Document
+                        {
+                            TargetId = "id1"
+                        },
+                        new Document
+                        {
+                            TargetId = "id2"
+                        }
+                    }
+                    };
+                    doc2 = new Document
+                    {
+                        Deleted = false,
+                        SubDocuments = new List<Document>
+                    {
+                        new Document
+                        {
+                            TargetId = "id4"
+                        },
+                        new Document
+                        {
+                            TargetId = "id5"
+                        }
+                    }
+                    };
+                    doc3 = new Document
+                    {
+                        Deleted = true
+                    };
+
+                    session.Store(doc1);
+                    session.Store(doc2);
+                    session.Store(doc3);
+                    session.SaveChanges();
+                }
+            }
+
+            [Fact]
+            public void CanProjectWithArrayParameters()
+            {
+                using (var store = GetDocumentStore())
+                {
+                    SetUp(store);
+
+                    using (var session = store.OpenSession())
+                    {
+                        var ids = new[] { doc1.Id, doc2.Id, doc3.Id };
+                        string[] targetIds = { "id2" };
+
+                        var projection =
+                            from d in session.Query<Document>().Where(x => x.Id.In(ids))
+                            where d.Deleted == false
+                            select new
+                            {
+                                Id = d.Id,
+                                Deleted = d.Deleted,
+                                Values = d.SubDocuments
+                                        .Where(x => targetIds.Length == 0 || targetIds.Contains(x.TargetId))
+
+                                          .Select(x => new Result
+                                          {
+                                                  TargetId = x.TargetId,
+                                                  TargetValue = x.TargetValue
+                                          })
+                            };
+
+                        Assert.Equal("FROM Documents as d WHERE (id() IN ($p0)) AND (Deleted = $p1) " +
+                                     "SELECT { Id : d.Id, Deleted : d.Deleted, " +
+                                     "Values : d.SubDocuments.filter(function(x){return ([\"id2\"]).length===0||[\"id2\"].indexOf(x.TargetId)>=0;}).map(function(x){return {TargetId:x.TargetId,TargetValue:x.TargetValue};}) }"
```

Comment:
Can we make this a single value, instead of repeating it twice?
So something like:
`filter(function(x){var array = [\"id2\"]; return (array).length===0||array.indexOf(x.TargetId)>=0;})